### PR TITLE
Move space convertion functions

### PR DESCRIPTION
**Pull request by phi-lira**
### Purpose of this PR
 Move Space Convertion functions from HDRP/LWRP/ShaderGraph to Core.

---
### Release Notes
 - Space convertion functions are now in SRP Core ShaderLibrary

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=move-space-convertion-functions&unity_branch=trunk

**Manual Tests**: Done by me.

**Automated Tests**: No need to add automated tests.

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: Medium. Mostly refactors/move.
**Halo Effect**: High. Changes all render pipelines and breaks current API.

---
### Comments to reviewers
Just moves functions to Core and make LWRP, HDRP and Shader Graph include them.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariablesFunctions.hlsl [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #ifndef UNITY_SHADER_VARIABLES_FUNCTIONS_INCLUDED
 #define UNITY_SHADER_VARIABLES_FUNCTIONS_INCLUDED
 
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/SpaceConvertionFunctions.hlsl"
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariablesFunctions.hlsl]: Suggestion: rename to SpaceConversions.hlsl or SpaceTransforms.hlsl , "Functions" is redundant in this context.

File: com.unity.shadergraph/ShaderGraphLibrary/ShaderVariablesFunctions.hlsl [hunk 2]
```diff
@@ -1,76 +1,7 @@
...
-    return normalize(mul(normalOS, (float3x3)GetWorldToObjectMatrix()));
-#endif
-}
-
-// Tranforms position from world space to homogenous space
-float4 TransformWorldToHClip(float3 positionWS)
-{
-    return mul(GetWorldToHClipMatrix(), float4(positionWS, 1.0));
-}
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/SpaceTransforms.hlsl"
```
Comment (pbbastian) [inline com.unity.shadergraph/ShaderGraphLibrary/ShaderVariablesFunctions.hlsl]: Shader Graph doesn't have a dependency on `com.unity.render-pipelines.core` right now, but you should add it in this PR :)

**Comment by pbbastian** [comment]
Other than my comments, the changes looks good to me. The only difference I can see between what used to be in `ShaderGraphLibrary/ShaderVariablesFunctions.hlsl` and what is now in `ShaderLibrary/SpaceTransforms.hlsl` is that some things are now `real` instead of `float`, which should be fine.

**Comment by phi-lira** [comment]
Implemented the requested review. The change from float to real was intentional. @pbbastian I've got errors in shader graph and had to reimport those. Do I have to bump a version for this?

**Pull request merged**